### PR TITLE
Will/eng 1420/revise analysis

### PR DESCRIPTION
**Pull request by wbrinkerhoff**
We might want to add something in the readme about doing a revision, but want to double check how much we should publicize the feature.

### CODE REVIEW

File: tests/test_analysis.py [hunk 1]
```diff
@@ -1412,3 +1413,94 @@ def test_analysis_upload_file_makes_expected_calls_s3(
...
+    analysis_data["latest_revision_model_state"] = "T"
+    analysis = Analysis(patched_fero_client, analysis_data)
+    assert analysis.is_retraining() is False
+    patched_fero_client.get.assert_called_with(
+        f"/api/analyses/{analysis_data['uuid']}/"
+    )
+
+
+def test_is_retraining_retraining(analysis_data, patched_fero_client):
+    """Test that is_retraining() returns false if the analysis is training"""
```
Comment (lukegrecki) [inline tests/test_analysis.py:1431]: This should read `is_retraining() returns true` I think.

File: tests/test_analysis.py [hunk 2]
```diff
@@ -1412,3 +1413,94 @@ def test_analysis_upload_file_makes_expected_calls_s3(
         [mock.call({"upload_type": "s3", "other_info": "test_value"}, "test_tag", fp)]
     )
     analysis._client._azure_upload.assert_has_calls([])
+
+
+def test_is_retraining_not_retraining(analysis_data, patched_fero_client):
+    """Test that is_retraining() returns false if the analysis is not training"""
+
+    patched_fero_client.get.return_value = analysis_data.copy()
+    analysis_data["latest_revision_model_state"] = "T"
```
Comment (lukegrecki) [inline tests/test_analysis.py:1422]: Shouldn't this status be an `"I"` or something other than training based on the description of the test?

**Review by lukegrecki — CHANGES_REQUESTED** [review]
Looks good! Two minor comments related to the tests.

File: tests/test_analysis.py (hunk 2 above)
Comment (wbrinkerhoff) [reply tests/test_analysis.py:1422]: We set that to make sure it's actually reloading based on the latest status of analysis, not the current cached value.

Basically, it covers the path like this.
```python
analysis.revise()  # internal data is 'T'
analysis.is_retraining()  # would still be training if we don't refresh before checking
```

File: tests/test_analysis.py (hunk 1 above)
Comment (wbrinkerhoff) [reply tests/test_analysis.py:1431]: yes.

**Review by lukegrecki — APPROVED** [review]

**Pull request merged**